### DISCUSSION TITLE
why can't I access this attribute?

### DISCUSSION THREAD

**Discussion by dotysan** [PyAV-Org/PyAV#1606] [1. Help] [answered]
```Python
    input_container = av.open(RTSP_URL)
    video_stream: av.VideoStream = input_container.streams.video[0]
    print(video_stream)
    print(video_stream.gop_size)
```
```
<av.VideoStream #1 h264, yuv420p 1280x720 at 0x7ff308dc4a00>
Traceback (most recent call last):
  File "test.py", line 28, in main
    print(video_stream.gop_size)
          ^^^^^^^^^^^^^^^^^^^^^
  File "av/video/stream.pyx", line 21, in av.video.stream.VideoStream.__getattr__
  File "av/video/codeccontext.pyx", line 160, in av.video.codeccontext.VideoCodecContext.gop_size.__get__
RuntimeError: Cannnot access 'gop_size' as a decoder
```

Shouldn't it be a simple int? https://github.com/PyAV-Org/PyAV/blob/ee30e4f24a49bf101072b751b0c39df14d36b851/av/video/stream.pyi#L33

**Comment by WyattBlue** [comment, marked as answer]
Because the `gop_size` value for a decoder would, at best, produce a nonsense answer, and at worst, be slightly wrong or coincidentally correct.

`gop_size` tells the encoder how many non-keyframes there should be between each keyframe. Of course, the encoder might use that as only a rough guideline or ignore it completely. If you want to know the "gop_size" for a decoder. You can decode the frames in a loop, then check `frame.is_keyframe` to calculate the value.

**Comment by dotysan** [comment]
Thank you!
